pkg/registry/core/shoot/strategy.go

```diff
@@ -171,7 +171,7 @@ func mustIncreaseGenerationForSpecChanges(oldShoot, newShoot *core.Shoot) bool {
 
 func removeDuplicateExtensions(shoot *core.Shoot) {
```

Comment:
Doesn't this still change order because of the sorting and the iteration over the map?

Reply:
Order will be changed(Sorted) only if duplicates are found else not. This is to ensure that after removing duplicates same config is applied every time.